### ISSUE TITLE
Reactions Block: add option to customize and / or remove the heading

### ISSUE THREAD

**Issue by jeherve** [Automattic/wordpress-activitypub#1096]
### What

It would be nice if we could customize and / or remove the heading of the Reactions block.

On my site, I would like to be able to:

1. Use the heading level I want
2. Change the text
3. Customize the display of the heading (top border, style, ...)
4. Remove it entirely if necessary

<img width="1317" alt="Image" src="https://github.com/user-attachments/assets/f6373b03-4f40-46a8-b38f-3158d4fb572f" />

### Why

The more customizable the block is, the easier it will be for folks to integrate it on their sites.

### How

Would it be possible to auto-insert a core group block instead, and that group block would have a heading block and a Reactions block? This way folks would then be able to customize it to their liking.

**Comment by jeherve** [comment]
cc @mattwiebe

**Comment by mattwiebe** [comment]
Agreed these would be nice! You should be able to edit the text, including an empty string that will remove it

**Comment by jeherve** [comment]
If that can help, in Jetpack's Related Posts block we relied on `InnerBlocks` to allow a heading block before the actual related posts element:
https://github.com/Automattic/jetpack/blob/a9966bf7dd5c6d41b87fa18f4f6be8f02a6e2ef7/projects/plugins/jetpack/extensions/blocks/related-posts/edit.js#L262-L265

Maybe we could do something similar with the Reactions block?

**Comment by vk17-starlord** [comment]
@obenland I came across this open issue and would like to contribute by working on a solution. Would it be alright if I take a look at it and submit a potential fix?

**Comment by pfefferle** [comment]
@vk17-starlord sure, give it a try :)

**Comment by vk17-starlord** [comment]
👋 Hi all, I’ve been exploring the Reactions block while looking into this enhancement. When I added the Reactions block to a page and viewed it on the frontend, nothing appeared — I couldn’t see the block rendered.

Before diving further, I wanted to check:

* Are there any specific requirements or configurations needed for the Reactions block to be visible on the frontend?
* Also, I noticed that on the editor side, the reactions seem to be dummy placeholders. Could you please clarify where the actual Fediverse reactions are fetched from on the frontend? Is there a background process or specific condition that triggers fetching/display?

Any guidance would be greatly appreciated. Thanks in advance!

Editor screenshot 

<img width="1447" alt="Image" src="https://github.com/user-attachments/assets/867bb3ee-9391-4df8-b309-97878361309f" />

Frontend Screenshot 

<img width="1469" alt="Image" src="https://github.com/user-attachments/assets/f16d91ea-c24c-432f-92d2-5192288e3b94" />

@pfefferle

**Comment by vk17-starlord** [comment]
> If that can help, in Jetpack's Related Posts block we relied on `InnerBlocks` to allow a heading block before the actual related posts element: https://github.com/Automattic/jetpack/blob/a9966bf7dd5c6d41b87fa18f4f6be8f02a6e2ef7/projects/plugins/jetpack/extensions/blocks/related-posts/edit.js#L262-L265
> 
> Maybe we could do something similar with the Reactions block?

@jeherve @pfefferle  based on your request I have implemented the inner blocks for heading here is the working 

https://www.loom.com/share/38c995e3db104592a63f386f26c38a1b?sid=c6a457c8-b016-4f82-90f6-24181b033015 

please review and let me know your thoughts about this
